src/main/java/rx/internal/operators/OperatorGroupBy.java

```diff
@@ -219,6 +219,13 @@ public void onCompleted() {
             if (done) {
                 return;
             }
+            List<GroupedUnicast<K, V>> list = new ArrayList<GroupedUnicast<K, V>>(groups.values());
+            groups.clear();
+            
+            for (GroupedUnicast<K, V> e : list) {
```

Comment:
Why not just iterate over `groups.values()` here and avoid `list` allocation? 

Or you want array structure for better performance? If so, not sure you're winning something big because you're iterating over `values()` in the constructor of `ArrayList` and then here.